File: chatbot-rag/services/session_manager.py
from fastapi import WebSocket
from typing import Dict, Any, List

class VoiceSession:
    def __init__(self, connection_id: str, ws: WebSocket, tenant_id: str, user_id: str, session_id: str = None):
        self.connection_id = connection_id
        self.ws = ws
        self.tenant_id = tenant_id
        self.user_id = user_id
        self.session_id = session_id
        self.audio_chunks: List[bytes] = []
        self.is_processing = False

class SessionManager:
    def __init__(self):
        self._sessions: Dict[str, VoiceSession] = {}

    def register(self, connection_id: str, ws: WebSocket, tenant_id: str, user_id: str, session_id: str = None) -> VoiceSession:
        session = VoiceSession(connection_id, ws, tenant_id, user_id, session_id)
        self._sessions[connection_id] = session
        return session

    def get(self, connection_id: str) -> VoiceSession:
        return self._sessions.get(connection_id)

    def remove(self, connection_id: str) -> None:
        if connection_id in self._sessions:
            del self._sessions[connection_id]

    def append_audio(self, connection_id: str, chunk: bytes) -> None:
        session = self.get(connection_id)
        if session:
            session.audio_chunks.append(chunk)

    def clear_audio(self, connection_id: str) -> None:
        session = self.get(connection_id)
        if session:
            session.audio_chunks = []
            session.is_processing = False

session_manager = SessionManager()

File: voice-gateway-legacy/src/services/sessionManager.ts
import { WebSocket } from 'ws';

export interface VoiceSession {
  ws: WebSocket;
  tenantId: string;
  userId: string;
  sessionId?: string;
  audioChunks: Buffer[];
  isProcessing: boolean;
}

export class SessionManager {
  private sessions: Map<string, VoiceSession> = new Map();

  /**
   * Registers a new voice session
   */
  public register(id: string, ws: WebSocket, tenantId: string, userId: string, sessionId?: string): VoiceSession {
    const session: VoiceSession = {
      ws,
      tenantId,
      userId,
      sessionId,
      audioChunks: [],
      isProcessing: false,
    };
    this.sessions.set(id, session);
    return session;
  }

  /**
   * Retrieves a session by ID
   */
  public get(id: string): VoiceSession | undefined {
    return this.sessions.get(id);
  }

  /**
   * Removes a session
   */
  public remove(id: string): void {
    this.sessions.delete(id);
  }

  /**
   * Appends audio data to the session's buffer
   */
  public appendAudio(id: string, data: Buffer): void {
    const session = this.sessions.get(id);
    if (session) {
      session.audioChunks.push(data);
    }
  }

  /**
   * Clears accumulated audio data and resets processing state
   */
  public clearAudio(id: string): void {
    const session = this.sessions.get(id);
    if (session) {
      session.audioChunks = [];
      session.isProcessing = false;
    }
  }

  /**
   * Sets processing flag
   */
  public setProcessing(id: string, processing: boolean): void {
    const session = this.sessions.get(id);
    if (session) {
      session.isProcessing = processing;
    }
  }
}
export const sessionManager = new SessionManager();
